pkg/apis/configuration/validation/validation.go

```diff
@@ -815,3 +816,23 @@ func rejectPlusResourcesInOSS(upstream v1alpha1.Upstream, idxPath *field.Path, i
 
 	return allErrs
 }
+
+func validateQueue(queue *v1alpha1.UpstreamQueue, fieldPath *field.Path, isPlus bool) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if queue == nil {
+		return allErrs
+	}
+
+	if !isPlus {
+		allErrs = append(allErrs, field.Forbidden(fieldPath, "queue is only supported in NGINX Plus"))
+		return allErrs
+	}
+
+	allErrs = append(allErrs, validateTime(queue.Timeout, fieldPath.Child("timeout"))...)
+	if queue.Size <= 0 {
+		allErrs = append(allErrs, field.Required(fieldPath.Child("size"), "must be positive and greater than 0"))
```

Comment:
```suggestion
		allErrs = append(allErrs, field.Required(fieldPath.Child("size"), "must be positive"))
```

I think `"must be positive and greater than 0"` sounds redundant, what about `"must be positive"` instead? a positive number is already greater than 0 so I don't think the second part adds any information. Also for consistency (eg `allErrs = append(allErrs, field.Invalid(fieldPath.Child("number"), buff.Number, "must be positive"))`